File: BookManagementSystem/EBook.java
package BookManagementSystem;

// The Ebook class must also inherit the class book and have an additional attribute:
//      sizeInMb: a double representing the size of the ebook file in megabytes.

public class EBook extends Book {
    private double sizeInMb;

    // constructor 
    public EBook(String title, String author, String isbn, double price, double sizeInMb) {
        super(title, author, isbn, price);
        this.sizeInMb = sizeInMb;
    }

    // getters and setters
    public double getSizeInMb() {
        return sizeInMb;
    }
    public void setSizeInMb(double sizeInMb) {
        this.sizeInMb = sizeInMb;
    }
}

File: BMSRust/src/EBook.rs
mod Book;
use Book::Book;

pub struct EBook{
    book: Book,
    size_in_mb: f64,
}

pub impl EBook{
    pub fn new(title: String, author: String, isbn: String, price: f64, size_in_mb: f64) -> Self{
        EBook{
            book: Book{
                title,
                author,
                isbn,
                price,
            }, 
            size_in_mb
        }
    }

    // getters and setters

    // getter for size_in_mb
    pub fn get_size_in_mb(&self) -> &f64{
        &self.size_in_mb
    }
    // setter for size_in_mb
    pub fn set_size_in_mb(&mut self, size_in_mb: f64){
        self.size_in_mb = size_in_mb;
    }
}